### PR TITLE
Added support for invoices and add some additional data in client and lineitems

### PR DESCRIPTION
**Pull request by chaelli**

### CODE REVIEW

File: Harvest.Api/HarvestClient.cs [hunk 1]
```diff
@@ -166,6 +166,12 @@ public bool IsRedirectUri(Uri uri)
 
             return uri.GetLeftPart(UriPartial.Path) == this.RedirectUri.GetLeftPart(UriPartial.Path);
         }
+
+        public void SetDefaultAccountId(long accountId)
```
Comment (zVolodymyr) [inline Harvest.Api/HarvestClient.cs]: DefaultAccountId already has public setter

File: Harvest.Api/HarvestClient.cs [hunk 2]
```diff
@@ -285,7 +291,54 @@ await SimpleRequestBuilder($"{harvestApiUrl}/time_entries/{entryId}", accountId,
                 .SendAsync<ProjectsResponse>(_httpClient, cancellationToken);
         }
 
-        public async Task<TasksResponse> GetTasksAsync(DateTime? updatedSince = null, int? page = null, int? perPage = null,
+        public async Task<Project> GetProjectAsync(long? projectId, bool includeTasks = false, long? accountId = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await RefreshTokenIsNeeded();
+            var project = await SimpleRequestBuilder($"{harvestApiUrl}/projects/{projectId}", accountId)
+                .SendAsync<Project>(_httpClient, cancellationToken);
+            if (includeTasks)
```
Comment (zVolodymyr) [inline Harvest.Api/HarvestClient.cs]: task_assigments should be moved to separate method

File: Harvest.Api/HarvestClient.cs [hunk 3]
```diff
@@ -285,7 +291,54 @@ await SimpleRequestBuilder($"{harvestApiUrl}/time_entries/{entryId}", accountId,
...
+                .Query("client_id", clientId)
+                .SendAsync<Invoice>(_httpClient, cancellationToken);
+        }
+
+        // using "customerId" because clientId is already in use
+        public async Task<Client> GetClientAsync(string customerId, long? clientId = null, long? accountId = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await RefreshTokenIsNeeded();
+            return await SimpleRequestBuilder($"{harvestApiUrl}/clients/{customerId}", accountId)
+                .Query("client_id", clientId)
```
Comment (zVolodymyr) [inline Harvest.Api/HarvestClient.cs]: Can't find client_id query parameter for clients/{id} API

File: Harvest.Api/Models/IdNameModel.cs [hunk 4]
```diff
@@ -6,7 +6,7 @@ namespace Harvest.Api
 {
     public class IdNameModel
     {
-        public long Id { get; set; }
+        public long? Id { get; set; }
```
Comment (zVolodymyr) [inline Harvest.Api/Models/IdNameModel.cs]: It affect a lot of models, can you provide example when Id is null?

Comment (chaelli) [reply Harvest.Api/Models/IdNameModel.cs]: could not replicate it - made it "long" again

**Comment by zVolodymyr** [comment]
will refactor it later

**Pull request merged**